astropy/units/core.py

```diff
@@ -1757,7 +1758,13 @@ class NamedUnit(UnitBase):
         If any of the given unit names are not valid Python tokens.
     """
 
-    def __init__(self, st, doc=None, format=None, namespace=None):
+    def __init__(
+        self,
+        st: str | list[str] | tuple[list[str], list[str]],
+        doc: str | None = None,
+        format: Mapping[str, str] | None = None,
+        namespace: MutableMapping[str, object] | None = None,
+    ) -> None:
         UnitBase.__init__(self)
```

Comment:
Might as well `super()` this if we are adding small changes.

Reply:
Agreed!

Reply:
Should I push a second commit with code changes or should we keep this pull request for type annotations only and make the code changes in a separate pull request?

Reply:
2nd commit. We rarely do cleanup PRs, so it's good to clean up small things as we encounter them.